src/MagnumPlugins/DrFlacAudioImporter/DrFlacImporter.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016
+              Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2015 Jonathan Hale <squareys@googlemail.com>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "DrFlacImporter.h"
+
+#include <Corrade/Utility/Assert.h>
+#include <Corrade/Utility/Debug.h>
+#include <Corrade/Utility/Endianness.h>
+
+#define DR_FLAC_IMPLEMENTATION
+#include "dr_flac.h"
+
+namespace Magnum { namespace Audio {
+
+DrFlacImporter::DrFlacImporter() = default;
+
+DrFlacImporter::DrFlacImporter(PluginManager::AbstractManager& manager, std::string plugin): AbstractImporter(manager, std::move(plugin)) {}
+
+auto DrFlacImporter::doFeatures() const -> Features { return Feature::OpenData; }
+
+bool DrFlacImporter::doIsOpened() const { return _data; }
+
+void DrFlacImporter::doOpenData(Containers::ArrayView<const char> data) {
+
+    drflac* handle = drflac_open_memory(reinterpret_cast<const UnsignedByte*>(data.data()), data.size());
+
+    if(handle == NULL) {
+        Error() << "Audio::DrFlacImporter::openData(): failed to open and decode FLAC data";
+        return;
+    }
+
+    _handle = (void*)handle;
+
+    int32_t* decodedData = handle->pDecodedSamples;
+    uint64_t samples = handle->totalSampleCount;
+    uint32_t frequency = handle->sampleRate;
+    uint8_t numChannels = handle->channels;
+    uint8_t bitsPerSample = handle->bitsPerSample;
+
+    _frequency = frequency;
+
+    Containers::Array<char> tempData{reinterpret_cast<char*>(decodedData), size_t(samples*numChannels*2)};
```

Comment:
@mosra Question I wasn't easily able to resolve myself; does `Containers::Array<char>` copy `decodedData`, or is there a way to make it do so? Then we could get rid of the entire `_handle`, and I wouldn't have to manage it at all.